docs/ANALYTICS.md

```diff
@@ -0,0 +1,98 @@
+# Analytics
+
+This document describes best practices and styles for using Google Analytics.
+
+## Tracking Events
+
+Use a spreadsheet to track all events that are tracked in the app.
+You can use [this template](https://docs.google.com/spreadsheets/d/1yEgTKGOUVgUlfWrG-qjul5ers80qcINw4UcisZoMSD8/edit?usp=sharing) containing all recommended events.
+
+### Style Guide
+
+- Event names must be in `snake_case`.
+- Parameter names must be in `snake_case`.
+- Parameter values should be in `snake_case`, but may be in other formats if necessary.
+
+### Recommended Events
+
+The following events are recommended for all apps:
+
+| Event Name | Parameters | Description |
+| ---------- | ---------- | ----------- |
+| `app_open` | | The app is opened. |
+| `login` | `method` | The user logs in. |
+| `sign_up` | `method` | The user signs up. |
+| `delete_account` | | The user deletes their account. |
+| `logout` | | The user logs out. |
+| `share` | `content_type`, `item_id` | The user shares content. |
+| `search` | `search_term` | The user searches for content. |
+| `url_view` | `url` | The user views a URL. |
+| `screen_view` | `firebase_screen`, `firebase_screen_class` | The user views a screen. |
+| `campaign_details` | `source`, `medium`, `campaign` | The user opens the app with a link containing UTM parameters. |
+
+Some of the above events are already defined in the Google Analytics SDK. You can see the full list of pre-defined GA events [here](https://support.google.com/analytics/answer/9267735?hl=en).
+
+Other events are automatically collected by the Google Analytics SDK. You can see the full list [here](https://support.google.com/analytics/answer/9234069?hl=en).
+
+### Custom Events
+
+Before you create a custom event, make sure the event you want to create isn't already collected through an automatically collected event or recommended as a recommended event. It's always better to use an existing event because these events automatically populate dimensions and metrics that are used in your reports.
+
+The event name should describe what you intend to measure with the event. For example, if you're measuring donations, the name might be "donate".
+
+The event parameters of a custom event provide more information about the action that took place.
```

Comment:
Some parameters populate [existing dimensions and metrics](https://support.google.com/analytics/answer/9143382) from Google Analytics. Before creating a new parameter, make sure there is no existing option that can match the information you want to send. In case you need to create a custom parameter, because there is no dimension or metric provided for your event parameter, you must create a [custom dimension or metric](https://support.google.com/analytics/answer/10075209?) in order to access the data in the reports. Be aware of the limits on the numbers of custom dimensions an metrics that you can create, documented [here](https://support.google.com/analytics/answer/10075209?sjid=3848261877583570628-EU).